python/cmake/kvikio_python_helpers.cmake

```diff
@@ -0,0 +1,56 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+#[=======================================================================[.rst:
+add_cython_modules
+------------------
+
+Generate C(++) from Cython and create Python modules.
+
+.. code-block:: cmake
+
+  add_cython_modules(<ModuleName...>)
+
+Creates a Cython target for a module, then adds a corresponding Python
+extension module.
+
+``ModuleName``
+  The list of modules to build.
+
+#]=======================================================================]
+function(add_cython_modules cython_modules)
+  foreach(cython_module ${cython_modules})
+    add_cython_target(${cython_module} CXX PY3)
+    add_library(${cython_module} MODULE ${cython_module})
+    python_extension_module(${cython_module})
+
+    # To avoid libraries being prefixed with "lib".
+    set_target_properties(
+        ${cython_module}
+        PROPERTIES PREFIX ""
+                   CXX_STANDARD 17
```

Comment:
Do we need C++17 here?

Reply:
My assumption was yes because we used to specify it in the [`extra_compile_args`](https://github.com/rapidsai/kvikio/pull/32/files#diff-eb8b42d9346d0a5d371facf21a8bfa2d16fb49e213ae7c21f03863accebe0fcfL93). I assumed that it would be necessary because we're compiling against headers that require C++17 (and therefore may use C++17 features).